src/Core/src/ViewExtensions.cs

```diff
@@ -27,6 +28,44 @@ namespace Microsoft.Maui
 {
 	public static partial class ViewExtensions
 	{
+		public static void DisconnectHandlers(this IView view)
+		{
+			// For our first go here
+			// My thinking is to build a flat list of all views in the tree
+			// And then iterate down the list disconnecting handlers.
+			// This gives me a stable list of views to call disconnecthandler on
+			// I'm assuming as this PR evolves we'll probably add some interfaces
+			// that allow handlers to manage their own children and disconnect flow
+			List<IView> _flatList = new List<IView>();			
+			BuildFlatList(view, _flatList);
+
+			foreach(var viewToDisconnect in _flatList)
+			{
+				viewToDisconnect.Handler?.DisconnectHandler();
+			}
+
+			void BuildFlatList(IView view, List<IView> flatList)
```

Comment:
What about returning an `IEnumerable<IView>` and use `yield` instead of filling up a list?

Reply:
yea, my concern is losing track of children because of the disconnect process, or the size of visual children changing for some reason. I use the flat list to try to avoid as many side effects as possible during the disconnect pass.

- Here's every element in the visual tree
- I am now going to call disconnect on all of them

As things disconnect the world begins to change, so, somewhat just trying to be sure that literally every handler will get it called.

Reply:
I see your point, then should we move `flatList.Add(view);` after `if (view is IVisualTreeElement vte)` in order to start the loop from the leafs?

Reply:
yea, I think we still need to go top down because the parent needs to have the ability to control the disconnect of the leaves. 

For example, the disconnect on the LayoutView can get confused if the leaves get disconnected before it can manage the cleanup. Same with CollectionView.

Worst case we'll have a parent that ends up calling disconnect on the children and this extension method will just do nothing on the children that don't have handlers.